core/node/consensus/src/storage/vm_reader.rs

```diff
@@ -0,0 +1,341 @@
+use std::time::Duration;
+
+use zksync_concurrency::ctx::Ctx;
+use zksync_contracts::load_contract;
+use zksync_node_api_server::{execution_sandbox::BlockStartInfo, tx_sender::TxSender};
+use zksync_system_constants::DEFAULT_L2_TX_GAS_PER_PUBDATA_BYTE;
+use zksync_types::{
+    api::BlockId,
+    ethabi::{Address, Contract, Function, Token},
+    fee::Fee,
+    l2::L2Tx,
+    transaction_request::CallOverrides,
+    Nonce, U256,
+};
+
+use crate::storage::ConnectionPool;
+
+pub struct VMReader {
+    pool: ConnectionPool,
+    tx_sender: TxSender,
+    consensus_authority_contract: Contract,
+    validator_registry_address: Option<Address>,
+    validator_registry_contract: Contract,
+    attester_registry_address: Option<Address>,
+    attester_registry_contract: Contract,
+}
+
+impl VMReader {
+    pub async fn new(
+        ctx: &Ctx,
+        block_id: BlockId,
+        pool: ConnectionPool,
+        tx_sender: TxSender,
+        consensus_authority_address: Address,
+    ) -> Self {
+        let consensus_authority_contract = load_contract("contracts/l2-contracts/artifacts-zk/contracts/ConsensusAuthority.sol/ConsensusAuthority.json");
+        let validator_registry_contract = load_contract("contracts/l2-contracts/artifacts-zk/contracts/ValidatorRegistry.sol/ValidatorRegistry.json");
+        let attester_registry_contract = load_contract("contracts/l2-contracts/artifacts-zk/contracts/AttesterRegistry.sol/AttesterRegistry.json");
```

Comment:
`zksync_contracts` crate currently serves (somewhat) as a schema for our FS dependencies, so probably it would be better to load the contracts there.
You can create a module for `consensus_l2_contracts` and load them via `pub fn load_consensus_l2_contracts() -> Option<ConsensusContracts>`.

Reply:
Fixed here: https://github.com/matter-labs/zksync-era/pull/2416/commits/5a925c6aced18514645823aa4f3332f1bee354f2